tests/test_public_cli.py

```
@@ -1400,3 +1400,71 @@ def test_plot_bad_nulls(
+    tmp_dir = Path(tmp_path)
+
+    with pytest.raises(
+        SystemExit, match="ERROR: Database /does/not/exist does not exist"
+    ):
+        public_cli.classify(database=Path("/does/not/exist"), outdir=tmp_dir)
+
+    with pytest.raises(
+        SystemExit, match="ERROR: Output directory /does/not/exist does not exist"
+    ):
+        public_cli.classify(database=":memory:", outdir=Path("/does/not/exist/"))
+
+    tmp_db = tmp_dir / "export.sqlite"
+    session = db_orm.connect_to_db(tmp_db)
+    with pytest.raises(SystemExit, match="ERROR: Database contains no runs."):
+        public_cli.classify(database=tmp_db, outdir=tmp_dir)
+
+    config = db_orm.db_configuration(
+        session, "fastANI", "fastani", "1.2.3", create=True
+    )
+    db_orm.add_run(
+        session,
+        config,
+        cmdline="pyani fastani ...",
+        fasta_directory=Path("/does/not/exist"),
+        status="Empty",
+        name="Trial A",
+    )
+    db_orm.add_run(
+        session,
+        config,
+        cmdline="pyani fastani ...",
+        fasta_directory=Path("/does/not/exist"),
+        status="Empty",
+        name="Trial B",
+    )
+    with pytest.raises(SystemExit, match="ERROR: Database has no run-id 3."):
+        public_cli.classify(database=tmp_db, outdir=tmp_dir, run_id=3)
+    with pytest.raises(
+        SystemExit,
+        match="ERROR: run-id 1 has no comparisons",
+    ):
+        public_cli.classify(database=tmp_db, outdir=tmp_dir, run_id=1)
+    # Should default to latest run, run-id 2
+    with pytest.raises(
+        SystemExit,
+        match="ERROR: run-id 2 has no comparisons",
+    ):
+        public_cli.classify(database=tmp_db, outdir=tmp_dir)
+    tmp_db.unlink()
+
+
+def test_classify_db(
+    capsys: pytest.CaptureFixture[str], input_genomes_tiny: Path, tmp_path: Path
+) -> None:
+    """Check classify working example."""
+    tmp_dir = Path(tmp_path)
+    db = input_genomes_tiny / "viral_database.db"
```

We should avoid having binary blobs like SQLite databases under version control. See some of the other tests which create a DB in code using mock values - or just call something very fast like sourmash or fastANI on the three viruses as part of this test?

---

I tried to copy the approach you used in tes_resume_partial_fastani, where you set up a database in code with mock values, but I ran into a FAILED tests/test_public_cli.py::test_partial_run - RecursionError: maximum recursion depth exceeded error. I'll take another look—maybe I missed something...

---

Recursion? That was unexpected from setting up a mock DB.

---

Sorry for not being clear. Setting up a mock database didn’t cause the error. The error occurred when I tried to run `classify` on that mock database. I’m assuming the recursion error happened because I use a recursive function to find cliques? I will have another look and see what happens... If I replicate the error, I will try to share it here...

---

I don't know how I missed it yesterday, but the `RecursionError` is raised when there is information missing in the DB. I accideltly only added values for identity and not the covearge 😥... I will need to optimise `pyani classify` to deal with diffrent scenarios, but it might be best to do it on a separate branch...